sov-modules/sov-state/src/backend.rs

```diff
@@ -42,9 +50,13 @@ impl<K: Encode, V: Encode + Decode, S: Storage> Backend<K, V, S> {
         )
     }
 
-    pub(crate) fn remove_value(&mut self, storage_key: StorageKey) -> Option<V> {
-        let storage_value = self.get_value(storage_key.clone())?;
-        self.storage.delete(storage_key);
+    pub(crate) fn remove_value(
```

Comment:
(non-blocking, out-of-scope): Should we add an additional `delete` method which doesn't return the old value? That way, we can avoid a read in the MPT if clients don't need it

Reply:
Created an issue: https://github.com/Sovereign-Labs/sovereign/issues/141